### PR TITLE
Homework5/Tests - Add Unit, Integration, Component & E2E Tests for Tracks

### PR DESCRIPTION
**Pull request by LiliaBilous**
## Summary

This PR adds full test coverage for the Track feature, including:

- **Unit tests** (Vitest)
  - `trackSchema.test.ts`: track schema validation
  - `trackQuerySchema.test.ts`: filter schema validation
  - `useAudioPlayer.test.ts`: audio player logic and edge cases
  - `audioStore.test.ts`: audio store state & actions

- **Integration tests**
  - `useFiltersWithUrl.test.ts`: filter and URL sync logic

- **Component test** (Playwright CT)
  - `TrackCard.spec.ts`: renders track info, emits `select`, shows waveform if `playing`

- **E2E test**
  - `createTrack.spec.ts`: simulates creating a new track (with route mocking)

## Commands

```
# Run all unit & integration tests
npm run test

# Run component tests (Playwright CT)
npm run test:ct

# Run E2E track tests
npm run test:e2e:track
```

## Notes

✅ All tests are passing
🔧 Uses `vi.fn()` and `expect()` for assertions
📊 HTML report available via:
  ```
  npx playwright show-report
  ```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,30 +2,36 @@
...
     "vue": "3.5.13",
     "vue-router": "4.5.0",
     "vue-toastification": "2.0.0-rc.5",
     "wavesurfer.js": "7.9.4",
     "zod": "3.25.36"
   },
   "devDependencies": {
+    "@pinia/testing": "^1.0.2",
+    "@playwright/experimental-ct-vue": "^1.53.1",
+    "@playwright/test": "^1.53.0",
```
Comment (dankozz1t) [inline package.json:29]: Новим пакетам також можна зафіксувати версію "^"

File: playwright-ct.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                },
+            },
+        },
+    },
+    projects: [
+        {
+            name: 'chromium',
+            use: { ...devices['Desktop Chrome'] },
+        },
+    ],
```
Comment (dankozz1t) [inline playwright-ct.config.ts:23]: Також можна ще додати мобілку
{
name: 'iphone',
use: { ...devices['iPhone 13'] },
},

File: test-results.json
Comment (dankozz1t) [inline test-results.json:1]: Я б це додав в гітІгнор

**Review by dankozz1t — APPROVED** [review]

File: tests/component/TrackCard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      onSelect,
+    },
+  })
+  await expect(component.getByTestId(`track-item-${mockTrack.id}-title`)).toHaveText(mockTrack.title)
+  await expect(component.getByTestId(`track-item-${mockTrack.id}-artist`)).toHaveText(`${mockTrack.artist} — ${mockTrack.album}`)
+
+  const checkbox = component.locator(`input[data-testid="track-checkbox-${mockTrack.id}"]`)
+  await expect(checkbox).not.toBeChecked()
+
+  await checkbox.click()
```
Comment (andriipriadko) [inline tests/component/TrackCard.spec.ts:33]: Після кліка можна додати `expect(checkbox).toBeChecked()`

**Review by andriipriadko — APPROVED** [review]
